testenv/FakeUserServiceApi.go

```diff
@@ -0,0 +1,141 @@
+//go:build !release
```

Comment:
@Rajagopalan-Ranganathan This build flag here ignores the file when the build is tagged with tag `release` which we do in `make binary` and elsewhere. Same as in https://github.com/RedHatInsights/authz/blob/main/infrastructure/repository/authzed/LocalSpiceDbContainer.go#L1 

That said, I just added another commit to rename the `testData` dir to `testdata`, as per the conventions this package is also ignored, see `go help packages`, last section:

> ...Directory and file names that begin with "." or "_" are ignored
by the go tool, as are directories named "testdata".